resources/troubleshooting-guides/push_notification_troubleshooting.md

```diff
@@ -0,0 +1,117 @@
+   2. To configuring the APNS certificate, register the app on Apple developer account and generate server certificate.  
+   For more information, see [Establishing a Certificate-Based Connection to APNs](https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/establishing_a_certificate-based_connection_to_apns?language=objc).
+   3. Verify that the APNS server certificate is configured in the Campaign instance.  
+   4. Verify this by clicking _Adobe Campaign -> Administrator -> Channels -> Mobile App (AEP SDK)_. Select the mobile app from the list and verify that push channel settings status for iOS is _iOS certficate with filename *.pem is successfully uploaded_.  
+ {% hint style="warning" %}
+ Verify that you are using production certificate for production version of app and debug certificate for debug version of app.
+ {% endhint %}
+   
+   ![alt text](../../.gitbook/assets/campaign_ios_cert_configured.png "Configured push certificate in Campaign.")
+   
+> For more information about configuration, see [Channel specific application configuration in Adobe Campaign.](https://helpx.adobe.com/campaign/kb/configuring-app-sdk.html#ChannelspecificapplicationconfigurationinAdobeCampaign).  
+  
+## To set up Android/iOS app for receiving push notifications  
+
+### To set up Android app  
+
+   1. Create the Firebase Messaging Service and add it to the Android Manifest file.  
+   For more information, see [Set up a Firebase Cloud Messaging client app on Android](https://firebase.google.com/docs/cloud-messaging/android/client)
+   2. Verify this step in Project Griffon.
+   For more information, see [Set up Project Griffon](https://aep-sdks.gitbook.io/docs/beta/project-griffon/set-up-project-griffon) and [Use Project Griffon](https://aep-sdks.gitbook.io/docs/beta/project-griffon/using-project-griffon).  
+   3. Generate a push token for app using FireBase Instance Id class.  
+   4. Set the push token to the Expereince Platform SDK by calling Set [Push Identifier](https://aep-sdks.gitbook.io/docs/using-mobile-extensions/adobe-campaign-standard/adobe-campaign-standard-api-reference#set-up-push-messaging).     
+
+### To set up iOS app  
+  
+  1. After you launch the app, register the app to receive the APNS token by calling `registerForRemoteNotifications`.  
+  This API generates the APNS token and returns token through the delegate function `didRegisterForRemoteNotificationsWithDeviceToken`.    
+  For more information, see [Registering Your App with APNs](https://developer.apple.com/documentation/usernotifications/registering_your_app_with_apns?language=objc)
+  2. In `didRegisterForRemoteNotificationsWithDeviceToken`, set the Push identifier by calling the `setPushIdentifier`.  
+  For more information, see [setPushIdentifier](https://aep-sdks.gitbook.io/docs/using-mobile-extensions/adobe-analytics-mobile-services#set-up-push-messaging).
```

Comment:
Lets add the extension name to the hyperlink name: `Mobile Services API reference - setPushIdentifier`. It will help differentiate from the other `setPushIdentifier` API.

Reply:
My mistake, this link should be the same one as I mentioned in my above comment: https://aep-sdks.gitbook.io/docs/using-mobile-extensions/mobile-core/identity/identity-api-reference#setPushIdentifierTitle with the hyperlink name of `Identity API reference - setPushIdentifier`.

Reply:
@rekrishn i see, its fine to leave it out then.

Reply:
thanks @rymorale!